modules/bmaas-setting-up-a-dedicated-namespace.adoc

```diff
@@ -0,0 +1,141 @@
+
+* Configure the external BMaaS hosts in a dedicated namespace separate from the {product-title} cluster hosts.
+* Create a dedicated namespace for BMaaS.
+* Set up a user with minimal privileges. 
+* Apply the necessary Role-Based Access Control (RBAC) roles.
+
+.Procedure
+
+. Create a dedicated namespace for BMaaS hosts by running the following command:
++
+[source,terminal]
+----
+$ oc new-project bmaas
+----
+
+. Create a BMaaS `bmadmin` user with the cluster's identity provider.
+
+. Create a secret in the `openshift-config` namespace with the identify provider by running the following command:
++
+[source,terminal]
+----
+$ oc create secret generic <identify_provider_arguments> -n openshift-config
+----
+
+. Configure OAuth to use the identity provider:
+
+.. Edit the OAuth resource by running the following command:
++
+[source,terminal]
+----
+$ oc edit oauth cluster
+----
+
+.. Add one of the following YAML snippets to the `spec.identityProviders` list:
++
+.Identity provider configuration examples
+[options="header"]
+|====
+|Type|Example
+| htpasswd
+a|[source,yaml]
+----
+- name: my_bmaas_provider
+  mappingMethod: claim
+  type: htpasswd
+  htpasswd:
+    fileData:
+      name: <secret>
+----
+| LDAP
+a|[source,yaml]
+----
+- name: my_bmaas_provider
+  mappingMethod: claim
+  type: ldap
+  ldap:
+    attributes:
+      id:
+      - dn
+      email:
+      - mail
+      name:
+      - cn
+      preferredUsername:
+      - uid
+----
+| GitHub
+a|[source,yaml]
+----
+- name: my_bmaas_provider
+  mappingMethod: claim
+  type: GitHub
+    github:
+      ca:
+        name: ca-config-map
+      clientID: {...}
+      clientSecret:
+        name: github-secret
+      hostname: ...
+      organizations:
+      - myorganization1
+      - myorganization2
+      teams:
+      - myorganization1/team-a
+      - myorganization2/team-b
+----
+|====
++
+For more information about identify providers, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.18/html-single/authentication_and_authorization/index[Authentication and authorization].
+
+
+.. Save and exit the editor.
+
+. Set the current namespace to `bmaas` by running the following command:
++
+[source,terminal]
+----
+$ oc project bmaas
+----
+
+. Clone the `baremetal-operator` repository to obtain the RBAC role definitions by running the following command:
++
+[source,terminal]
+----
+$ git clone -b release-4.19 https://github.com/openshift/baremetal-operator.git
+----
+
+. For each role you want to add, apply the appropriate RBAC role YAML file from the repository by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f baremetal-operator/config/base/rbac/<role_filename>.yaml
+----
+
+. Assign the `edit` role to the BMaaS `bmadmin` user in the `bmaas` namespace by running the following command:
++
+[source,terminal]
+----
+$ oc adm policy add-role-to-user edit bmadmin -n bmaas
```

Comment:
Move this to after `bmaas` namespace is created. It also puts the role creations and applying them to user bmadmin together (commands above and below this)